src/plugins/intel_npu/src/compiler/src/zero_compiler_in_driver.cpp

```diff
@@ -339,6 +333,60 @@ std::string LevelZeroCompilerInDriver<TableExtension>::serializeIOInfo(const std
            outputsPrecisionSS.str() + VALUES_SEPARATOR.data() + outputsLayoutSS.str();
 }
 
+template <typename TableExtension>
+void LevelZeroCompilerInDriver<TableExtension>::release(std::shared_ptr<const NetworkDescription> networkDescription) {
+    if (networkDescription->metadata.graphHandle != nullptr) {
+        ze_graph_handle_t graphHandle = static_cast<ze_graph_handle_t>(networkDescription->metadata.graphHandle);
+        auto result = _graphDdiTableExt->pfnDestroy(graphHandle);
+
+        if (ZE_RESULT_SUCCESS != result) {
+            _logger.error("Failed to release graph handle. L0 pfnDestroy result: %s, code %#X",
+                          ze_result_to_string(result).c_str(),
+                          uint64_t(result));
+        }
+    }
+}
+
+template <typename TableExtension>
+void LevelZeroCompilerInDriver<TableExtension>::fillCompiledNetwork(
```

Comment:
Consider making this a helper returning a vector and combine with metadata using ND ctor at the call site.
It's generally much more clear to pass immutable values instead of mutating them. Especially delegating object mutation to a method of other class make it harder to track the program state.